control-plane/catalog/to-consul/resource.go

```diff
@@ -739,67 +744,72 @@ func (t *ServiceResource) registerServiceInstance(
 			}
 		}
 		for _, endpoint := range endpointSlice.Endpoints {
-			for _, endpointAddr := range endpoint.Addresses {
-
-				var addr string
-				// Use the address and port from the Ingress resource if
-				// ingress-sync is enabled and the service has an ingress
-				// resource that references it.
-				if t.EnableIngress && t.isIngressService(key) {
-					addr = t.serviceHostnameMap[key].hostName
-					epPort = int(t.serviceHostnameMap[key].port)
-				} else {
-					addr = endpointAddr
-					if addr == "" && useHostname {
-						addr = *endpoint.Hostname
+			// Only consider endpoints that are ready
+			// nil represents an unknown state that can be interpreted as a non-terminating endpoint (assume ready or at least a state we shouldn't ignore)
+			// Ref: https://github.com/kubernetes/api/blob/5147c1a32f6a0b9b155bb84e59f933e0ff8a3792/discovery/v1/types.go#L128-L151
+			if endpoint.Conditions.Ready == nil || *endpoint.Conditions.Ready {
```

Comment:
@ndhanushkodi this is probably the part that I'd specifically like to callout that someone should review. I couldn't find any documented examples where `Ready|Terminating` would be nil and the edge cases that would entail so could remove this if preferred but the docs do imply this is "safe".

Reply:
> nil represents an unknown state that can be interpreted as a non-terminating endpoint (assume ready or at least a state we shouldn't ignore)

This makes sense to me, I'd lean towards keeping the nil check as you've done since it may be worth being safe and considering that endpoint rather than ignoring it.